TWTValidation/JSON Validator/Processors/TWTJSONSchemaPrettyPrinter.m

```diff
@@ -0,0 +1,279 @@
+- (void)processDependencyNode:(TWTJSONSchemaDependencyASTNode *)dependencyNode
+{
+    if (dependencyNode.valueSchema) {
+        [dependencyNode.valueSchema acceptProcessor:self];
+    } else {
+        // Node has a property set, which is an array of strings
+        [self pushNewObject:dependencyNode.propertySet];
+    }
+
+    [self setObject:[self popCurrentObject] forKey:dependencyNode.key];
+}
+
+
+#pragma mark - Schema-to-node conversion methods
+
+- (void)generateCommonSchemaFromNode:(TWTJSONSchemaASTNode *)node
+{
+    [self pushNewObject:[[NSMutableDictionary alloc] init]];
+    [self setObject:node.validTypes forKey:TWTJSONSchemaKeywordType];
+    [self setObject:node.schemaTitle forKey:TWTJSONSchemaKeywordTitle];
+    [self setObject:node.schemaDescription forKey:TWTJSONSchemaKeywordDescription];
+    [self setObject:node.validValues forKey:TWTJSONSchemaKeywordEnum];
+
+    [self setObject:[self schemaArrayFromNodeArray:node.andSchemas] forKey:TWTJSONSchemaKeywordAllOf];
+    [self setObject:[self schemaArrayFromNodeArray:node.orSchemas] forKey:TWTJSONSchemaKeywordAnyOf];
+    [self setObject:[self schemaArrayFromNodeArray:node.exactlyOneOfSchemas] forKey:TWTJSONSchemaKeywordOneOf];
+    [self setObject:[self schemaFromNode:node.notSchema] forKey:TWTJSONSchemaKeywordNot];
+
+    [self setObject:node.definitions forKey:TWTJSONSchemaKeywordDefinitions];
+}
+
+
+- (id)additionalItemsOrPropertiesFromNode:(TWTJSONSchemaASTNode *)additionalNode
+{
+    if (!additionalNode) {
+        return nil;
+    }
+
+    [additionalNode acceptProcessor:self];
+
+    return [self popCurrentObject];
+}
+
+
+- (NSDictionary *)dependencyDictionaryFromNodeArray:(NSArray *)array
+{
+    if (!array) {
+        return nil;
+    }
+
+    [self pushNewObject:[[NSMutableDictionary alloc] init]];
+    for (TWTJSONSchemaDependencyASTNode *node in array) {
+        [node acceptProcessor:self];
+    }
+
+    return [self popCurrentObject];
+}
+
+
+- (NSDictionary *)schemaFromNode:(TWTJSONSchemaASTNode *)node
+{
+    if (!node) {
+        return nil;
+    }
+
+    [node acceptProcessor:self];
+    return [self popCurrentObject];
+}
+
+
+- (NSArray *)schemaArrayFromNodeArray:(NSArray *)array
+{
+    if (!array) {
+        return nil;
+    }
+
+    NSMutableArray *nodeArray = [[NSMutableArray alloc] init];
+    for (TWTJSONSchemaASTNode *node in array) {
+        [node acceptProcessor:self];
+        [nodeArray addObject:[self popCurrentObject]];
+    }
+
+    return nodeArray;
+}
+
+
+- (NSDictionary *)schemaDictionaryFromKeyValuePairNodeArray:(NSArray *)array
+{
+    if (!array) {
+        return  nil;
+    }
+
+    [self pushNewObject:[[NSMutableDictionary alloc] init]];
+    for (TWTJSONSchemaKeyValuePairASTNode *node in array) {
+        [node acceptProcessor:self];
+    }
+
+    return [self popCurrentObject];
+}
+
+
+# pragma mark - Convenience methods
+
+- (void)setObject:(id)object forKey:(NSString *)key
```

Comment:
I don’t like this name. What do you think of `setObject:inCurrentObjectForKey:` or something like that.